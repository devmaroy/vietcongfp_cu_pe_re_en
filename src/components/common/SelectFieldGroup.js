import React from 'react';
import PropTypes from 'prop-types';

const SelectFieldGroup = ( { name, options, className, onChange, value, info } ) => {
    return (
        <div>
            <select
                name={ name }
                value={ value }
                className={ className }
                onChange={ onChange }
            >
                { 
                    options.length > 0 && (
                        options.map( ( option, index ) => <option key={ index } value={ option.key }>{ option.text }</option> )
                    )
                }
            </select>
            { info && <small>{ info }</small> }
        </div>
    );
};

SelectFieldGroup.propTypes = {
    name: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    value: PropTypes.string.isRequired,
    info: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string
};

export default SelectFieldGroup;