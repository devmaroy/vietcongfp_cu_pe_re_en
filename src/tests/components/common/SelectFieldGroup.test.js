import React from 'react';
import { shallow } from 'enzyme';
import SelectFieldGroup from '../../../components/common/SelectFieldGroup';

const options = [ 
    { text: 'Afghanistan', key: 'AF' }, 
    { text: 'Ã…land Islands', key: 'AX' }, 
    { text: 'Albania', key: 'AL' }, 
    { text: 'Algeria', key: 'DZ' } 
];

test( 'should correctly render SelectFieldGroup only with required props', () => {
    const wrapper = shallow(
            <SelectFieldGroup 
                name="countries"
                options={ options }
                value={ options[0].key }
                onChange={ () => {} }
            />
        );
    
    expect( wrapper ).toMatchSnapshot();
});

test( 'should correctly render TextFieldGroup with all props', () => {
    const wrapper = shallow( 
            <SelectFieldGroup 
                name="countries"
                options={ options }
                value={ options[0].key }
                onChange={ () => {} }
                info="Choose your country"
                className="select"
            /> 
        );

    expect( wrapper ).toMatchSnapshot();
});