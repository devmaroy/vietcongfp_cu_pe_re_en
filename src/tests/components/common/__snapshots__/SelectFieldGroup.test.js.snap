// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SelectFieldGroup only with required props 1`] = `
<div>
  <select
    name="countries"
    onChange={[Function]}
    value="AF"
  >
    <option
      key="0"
      value="AF"
    >
      Afghanistan
    </option>
    <option
      key="1"
      value="AX"
    >
      Åland Islands
    </option>
    <option
      key="2"
      value="AL"
    >
      Albania
    </option>
    <option
      key="3"
      value="DZ"
    >
      Algeria
    </option>
  </select>
</div>
`;

exports[`should correctly render TextFieldGroup with all props 1`] = `
<div>
  <select
    className="select"
    name="countries"
    onChange={[Function]}
    value="AF"
  >
    <option
      key="0"
      value="AF"
    >
      Afghanistan
    </option>
    <option
      key="1"
      value="AX"
    >
      Åland Islands
    </option>
    <option
      key="2"
      value="AL"
    >
      Albania
    </option>
    <option
      key="3"
      value="DZ"
    >
      Algeria
    </option>
  </select>
  <small>
    Choose your country
  </small>
</div>
`;
